src/components/figures/chart/chartJs/zoomable_chart/zoomable_chartjs.ts

```diff
@@ -0,0 +1,494 @@
+      } else if (Math.abs(position - xMax) < 5) {
+        startingPositionOnChart = xMin;
+      } else {
+        startingPositionOnChart = clip(position, left, right);
+      }
+      startX = this.computeCoordinate(startingPositionOnChart);
+    }
+    const originalXMin = this.store.originalAxisLimits[this.chartId].x!.min;
+    const originalXMax = this.store.originalAxisLimits[this.chartId].x!.max;
+
+    const computeNewAxisLimits = (position: number) => {
+      let xMin: number | undefined, xMax: number | undefined;
+      const { left, right } = this.slider!.chartArea;
+      if (this.mode === "moveInMaster") {
+        xMin = this.computeCoordinate(position - startingPositionOnChart)!;
+        if (xMin < originalXMin) {
+          xMin = originalXMin;
+        } else if (xMin > originalXMax - windowSize) {
+          xMin = originalXMax - windowSize;
+        }
+        xMax = xMin + windowSize;
+      } else if (this.mode === "selectInMaster") {
+        const upperBound = clip(position, left, right);
+        if (Math.abs(startingPositionOnChart - upperBound) > 5) {
+          const endX = this.computeCoordinate(upperBound);
+          if (startX === undefined || endX === undefined) {
+            return {};
+          }
+          xMin = Math.min(startX, endX);
+          xMax = Math.max(startX, endX);
+        }
+      }
+      return { min: xMin, max: xMax };
+    };
+
+    const onDragFromMasterChart = (ev: PointerEvent) => {
+      const position = ev.clientX - (this.sliderCanvas.el?.getBoundingClientRect().left ?? 0);
+      if (Math.abs(position - startingEventPosition) < 5) {
+        return;
+      }
+      const { min: xMin, max: xMax } = computeNewAxisLimits(position);
+      if (xMin !== undefined && xMax !== undefined) {
+        this.updateAxisLimits(xMin, xMax);
+      }
+    };
+
+    const onPointerUpInMasterChart = (ev: PointerEvent) => {
+      const position = ev.clientX - (this.sliderCanvas.el?.getBoundingClientRect().left ?? 0);
+      if (Math.abs(position - startingEventPosition) > 5) {
+        let { min: xMin, max: xMax } = computeNewAxisLimits(position);
+        if (xMin !== undefined && xMax !== undefined) {
+          if (!this.hasLinearScale) {
+            if (this.mode === "moveInMaster" && windowSize && !this.isBarChart) {
+              xMin = Math.round(xMin) - this.axisOffset;
+              xMax = xMin + windowSize;
+            } else {
+              xMin = Math.ceil(xMin) - this.axisOffset;
+              xMax = Math.floor(xMax) + this.axisOffset;
+            }
+          }
+          this.updateAxisLimits(xMin, xMax);
+        }
+      }
+      this.mode = undefined;
+      window.removeEventListener("pointermove", onDragFromMasterChart, true);
+      window.removeEventListener("pointerup", onPointerUpInMasterChart, true);
+    };
+
+    window.addEventListener("pointermove", onDragFromMasterChart, true);
+    window.addEventListener("pointerup", onPointerUpInMasterChart, true);
```

Comment:
this can lead to memory leaks if the component is unmounted before `removeEventListener` is called